### PR TITLE
(VDB-1050) Add missing indexes in public schema

### PR DESCRIPTION
**Pull request by Gslaughl**
Add the indexes that postgraphile complains about lacking when run with the `--no-ignore-indexes` flag

### CODE REVIEW

File: db/migrations/00016_add_receipts_fk_to_logs.sql [hunk 1]
```diff
@@ -8,8 +8,13 @@ ALTER TABLE full_sync_logs
             REFERENCES full_sync_receipts (id)
             ON DELETE CASCADE;
 
+CREATE INDEX full_sync_logs_receipt
```
Comment (Gslaughl) [inline db/migrations/00016_add_receipts_fk_to_logs.sql]: Have we switched to an append-only migration workflow in vDB yet? Migrations like this one make me think we have, but migrations have also been changed as recently as a week ago.

Comment (rmulhol) [reply db/migrations/00016_add_receipts_fk_to_logs.sql]: We have not, but migrations like this are a product of when we had made an attempt to do so earlier

File: db/migrations/00028_create_header_sync_logs_table.sql [hunk 2]
```diff
@@ -17,6 +17,13 @@ CREATE TABLE header_sync_logs
     UNIQUE (header_id, tx_index, log_index)
 );
 
+CREATE INDEX header_sync_logs_address
+    ON header_sync_logs (address);
```
Comment (rmulhol) [inline db/migrations/00028_create_header_sync_logs_table.sql]: Not introduced by this PR (nor something that needs to be addressed by it), but I'm wondering if `address` should be renamed to `address_id`

Comment (Gslaughl) [reply db/migrations/00028_create_header_sync_logs_table.sql]: Yeah that makes sense. I'll just do it in this PR

Comment (Gslaughl) [reply db/migrations/00028_create_header_sync_logs_table.sql]: Actually nvm, I didn't consider that would entail changes to the transformer repo as well 😅 but I'll take care of it later

**Pull request merged**